File: hw/vargas/u1/hw04MultiplicationTable/multiplicationTable.js
const readline = require("readline");

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

function askNumberTable() {
  rl.question("Enter a number from 1 to 12 to see its multiplication table: ", (input) => {
    const number = parseInt(input);
    if (isNaN(number) || number < 1 || number > 12) {
      console.log("Please enter a valid number in range of 1 and 12.");
      return askNumberTable();
    }

    console.log("===================================");
    console.log(`\nMultiplication table of number: ${number}`);
    for (let i = 1; i <= 12; i++) {
      const result = number * i;
      const numStr = String(number).padStart(2, " ");
      const iStr = String(i).padStart(2, " ");
      const resStr = String(result).padStart(3, " ");
      console.log(`|| ${numStr} x ${iStr} = ${resStr} ||`);
    }
    console.log("===================================");

    rl.question("Do you want to see another multiplication table? (y/n): ", (again) => {
      if (again.trim().toLowerCase() === "y") {
        askNumberTable();
      } else {
        rl.close();
      }
    });
  });
}
askNumberTable();

File: hw/vargas/u1/hw04MultiplicationTable/multiplicationTable.py
while True:
    while True:
        try:
            number = int(input("Enter a number from 1 to 12 to see its multiplication table:"))
            if 1 <= number <= 12:
                break  
            else:
                print("Please enter a valid number in range of 1 and 12.")
        except ValueError:
            print(" Please enter a valid integer.")
    print("===================================")
    print(f"\nMultiplication table of number:",number)
    for i in range(1, 13):
        result = number * i
        print(f"|| {number:2} x {i:2} = {result:3} ||")
    print("===================================")
    again = input("\nDo you want to see another multiplication table? (y/n): ").strip().lower()
    if again != 'y':
        break